Tryouts/Prototypes/Shell/MainWindow.xaml.cs

```diff
@@ -71,7 +72,7 @@ private void ShowChild_Click(object sender, RoutedEventArgs e)
         {
             var context = (sender as Button).DataContext;
```

Comment:
In this case the sender's `DataContext` is the same as the current Window's `DataContext`. But an even better approach would be to store the DataContext in a typed private field so that you wouldn't need to cast.